optimum/neuron/utils/misc.py

```diff
@@ -548,3 +549,60 @@ def check_if_weights_replacable(
         raise RuntimeError(
             "Unable to replace weights of the neuron model since its weights and neff are not separated, please set `inline_weights_to_neff=Talse` when converting the model to Neuron format."
         )
+
+
+# Copied and adapted from https://github.com/huggingface/optimum/blob/d03ab100206cb9f0e62167a36ee6997424bb9bb5/optimum/utils/save_utils.py#L27
+# To remove once we can bump to a transformers release including https://github.com/huggingface/transformers/pull/29169
+def maybe_load_preprocessors(
```

Comment:
Maybe put a check of the transformers version, if it's above the coming release, then fail. This way we can actually catch that and remove it.

Reply:
Sounds good!

Reply:
Just checked the PR I submited to transformers, it's not merged yet and I'm not 100% percent sure that it should be accepted by trfrs maintainers, thus completely don't know what trfr version to put here. Let's keep it this way, I will keep my eyes on how it goes...

Reply:
https://github.com/huggingface/transformers/pull/29169